educator/src/Dscp/Educator/Web/Educator/Queries.hs

```diff
@@ -281,6 +284,10 @@ educatorAddCertificate cert = do
         let faircv = privateBlockToFairCV blkHeader txs eAddr (defCertStudent, sName)
         pdf <- embedFairCVToCert (unReadyFairCV faircv) pdfRaw
 
+        _ <- runUpdate $ update (esSubmissions es)
+            ( \sub -> [srContentsHash sub <-. val_ (coerce pdfHash)])
+            $ \sub -> srHash sub `in_` (txs^..each.ptSignedSubmission.ssSubmission.idOf.to val_)
```

Comment:
Hm, I guess that if `contentsHash` changes then hash of the whole `Submission` changes as well which render database invalid :thinking:

Reply:
Eeh, of course this won't work well. Why do you even run this update? You create those submissions right here. You need to pass `pdfHash` to `addCertificateGrades` and put this hash in one of the newly created submissions there instead of choosing random hash (you might want to put it in the first submission, but not all of them, or you'll have duplicated transactions most likely).